api/src/shop/stripe_util.py

```diff
@@ -2,12 +2,134 @@
+) -> list[stripe.Price] | None:
+    try:
+        prices = list(retry(lambda: stripe.Price.list(product=stripe_product.stripe_id)))
+        if filterInactive:
+            prices = [p for p in prices if p["active"]]
+    except stripe.error.InvalidRequestError as e:
+        logger.warning(f"failed to retrive prices from stripe for stripe product with id {stripe_product.id}, {e}")
+        return None
+    for p in prices:
+        assert p.livemode == livemode
+    return prices
+
+
+def _create_stripe_product(makeradmin_product: Product, livemode: bool = False) -> stripe.Product | None:
+    id = f'prod_{makeradmin_product["id"]}' if livemode else f'test_{makeradmin_product["id"]}'
+    stripe_product = retry(
+        lambda: stripe.Product.create(
+            id=id,
+            name=makeradmin_product["name"],
+            description=f'Created by Makeradmin, product id (#{makeradmin_product["id"]})',
+        )
+    )
+    assert stripe_product.livemode == livemode
+    return stripe_product
+
+
+def _create_stripe_price(
+    makeradmin_product: Product, stripe_product: stripe.Product, priceType: PriceType, livemode: bool = False
+) -> stripe.Price | None:
+    id = f'prod_{makeradmin_product["id"]}_{priceType}' if livemode else f'test_{makeradmin_product["id"]}_{priceType}'
+    if "mån" in makeradmin_product["unit"] or "month" in makeradmin_product["unit"]:
+        interval = "month"
+    elif "år" in makeradmin_product["unit"] or "year" in makeradmin_product["unit"]:
+        interval = "year"
+    else:
+        raise RuntimeError(
+            f'Unexpected unit {makeradmin_product["unit"]} in makeradmin product {makeradmin_product["id"]}'
+        )
+    interval_count = makeradmin_product["smallest_multiple"]
+    recurring = {"interval": interval, "interval_count": interval_count}
+    stripe_price = retry(
+        lambda: stripe.Price.create(
+            nickname=id,
+            product=stripe_product.id,
+            unit_amount=convert_to_stripe_amount(makeradmin_product["price"] * interval_count),
+            currency=CURRENCY,
+            recurring=recurring,
+            metadata={"price_type": priceType},
+        )
+    )
+    assert stripe_price.livemode == livemode
+    return stripe_price
+
+
+def find_or_create_stripe_product(makeradmin_product: Product, livemode: bool = False) -> stripe.Product | None:
+    stripe_product = get_stripe_product(makeradmin_product, livemode)
+    if stripe_product is None:
+        return _create_stripe_product(makeradmin_product, livemode)
+    else:
+        if not stripe_product.active:
+            retry(lambda: stripe.Product.modify(stripe_product.id, active=True))
+        return stripe_product
+
+
+def _find_price_type(stripe_prices: list[stripe.Price] | None, price_type: PriceType):
+    if stripe_prices is None:
+        return None
+    for p in stripe_prices:
+        if p.metadata["price_type"] == price_type.value:
```

Comment:
You assume that this metadata exists. If it does not, makeradmin will crash.

Reply:
`p.metadata.get("price_type") == price_type.value` should do the trick if memory serves